sources/packages/backend/apps/api/src/route-controllers/education-program/education-program.institutions.controller.ts

```diff
@@ -113,6 +113,29 @@ export class EducationProgramInstitutionsController extends BaseController {
     );
   }
 
+  /**
+   * Allows a program to be deactivated.
+   * @param programId program to be deactivated.
+   * @param payload information to support the deactivation.
+   */
+  @ApiNotFoundResponse({
+    description: "Not able to find the education program.",
+  })
+  @ApiUnprocessableEntityResponse({
+    description: "The education program is already set as requested.",
+  })
+  @Patch(":programId/deactivate")
+  async deactivateProgram(
+    @Param("programId", ParseIntPipe) programId: number,
+    @UserToken() userToken: IInstitutionUserToken,
+  ): Promise<void> {
+    await this.educationProgramControllerService.deactivateProgram(
+      programId,
+      userToken.userId,
+      { institutionId: userToken.authorizations.institutionId },
```

Comment:
Do we not need to add the note when a institution user is causing the change?

Reply:
No, the notes are not provided when the institution calls the API, therefore they are not inserted.

Reply:
The overloads add as per the below comment will help to make it clear at the controller level.
https://github.com/bcgov/SIMS/pull/3023#discussion_r1532935272